tests/drhook/drhook_papi/CMakeLists.txt

```
@@ -106,4 +108,24 @@ ecbuild_add_executable( TARGET drhook_papi_user_counters
 ecbuild_add_test( TARGET fiat_test_drhook_papi_user_counters
         COMMAND drhook_papi_user_counters
         ENVIRONMENT DR_HOOK=1 DR_HOOK_OPT=COUNTERS DR_HOOK_PAPI_COUNTERS=PAPI_TOT_INS DR_HOOK_PROFILE=fiat_test_drhook_papi_user_counters
-        CONDITION HAVE_DR_HOOK_PAPI )
\ No newline at end of file
+        CONDITION HAVE_DR_HOOK_PAPI )
+
+set_tests_properties(fiat_test_drhook_papi_user_counters
+        PROPERTIES PASS_REGULAR_EXPRESSION "PAPI_TOT_INS" )
+
+# Test user specified counters going over max allowed
+
+ecbuild_add_executable( TARGET drhook_papi_user_counters_more_than_max
+  SOURCES drhook_papi_user_counters_more_than_max.F90
+  LIBS fiat
+  LINKER_LANGUAGE Fortran
+  CONDITION HAVE_DR_HOOK_PAPI
+  NOINSTALL )
+
+ecbuild_add_test( TARGET fiat_test_drhook_papi_user_counters_more_than_max
+        COMMAND drhook_papi_user_counters_more_than_max
+        ENVIRONMENT DR_HOOK=1 DR_HOOK_OPT=COUNTERS DR_HOOK_PAPI_COUNTERS=PAPI_TOT_CYC,PAPI_FP_OPS,PAPI_L1_DCA,PAPI_L2_DCM,PAPI_TOT_INS DR_HOOK_PROFILE=drhook_papi_user_counters_more_than_max
+        CONDITION HAVE_DR_HOOK_PAPI )
+
+set_tests_properties(fiat_test_drhook_papi_user_counters_more_than_max
```

This should be guarded with `if (TEST ... )`:
```suggestion
if (TEST fiat_test_drhook_papi_user_counters_more_than_max)
  set_tests_properties(fiat_test_drhook_papi_user_counters_more_than_max
endif()
```

---

I didn't realise this was a thing, it's much nicer 😁 Do you want me to add it to all `set_tests_properties` (including those from the merged NVTX feature)?